dashboard/app/models/user.rb

```diff
@@ -2261,6 +2259,40 @@ def providers
     end
   end
 
+  def num_failed_section_attempts
+    properties['section_attempts'] || 0
+  end
+
+  def reset_section_attempts?
+    num_failed_section_attempts == 0 || (DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0
+  end
+
+  # If 24 hours has passed, reset the section attempts value to zero.
```

Comment:
Update comment?  (Actually, it's probably a good idea to double-check the header comments for each of the functions below as well.  I think some of the logic was moved around but the header comments may not have been updated?)